bdk_rpc_example/src/main.rs

```diff
@@ -0,0 +1,182 @@
+                Ok(_) => log::info!("caught SIGTERM"),
+                Err(err) => log::warn!("failed to send SIGTERM: {}", err),
+            })
+            .expect("failed to set Ctrl+C handler");
+
+            // emit blocks thread
+            let join_handle =
+                std::thread::spawn(move || client.emit_blocks(&chan, &local_chain, initial_height));
+
+            for data in recv.iter() {
+                let mut update = ChainGraph::<TxHeight>::default();
+                let mut last_active_indexes = BTreeMap::new();
+
+                let txs = match data {
+                    RpcData::Blocks(blocks) => {
+                        for (&height, block) in blocks.iter() {
+                            update.insert_checkpoint(BlockId {
+                                height: height as _,
+                                hash: block.block_hash(),
+                            })?;
+                        }
+                        blocks
+                            .into_iter()
+                            .flat_map(|(height, block)| {
+                                block
+                                    .txdata
+                                    .into_iter()
+                                    .map(move |tx| (TxHeight::Confirmed(height as _), tx))
+                            })
+                            .collect::<Vec<_>>()
+                    }
+                    RpcData::Mempool(txs) => txs
+                        .into_iter()
+                        .map(|tx| (TxHeight::Unconfirmed, tx))
+                        .collect::<Vec<_>>(),
+                    RpcData::Stop(finished) => {
+                        log::info!("terminating... sync_finished={}", finished);
+                        drop(recv);
+                        break;
+                    }
+                };
+
+                for (height, tx) in txs {
+                    ensure_spk_cached(&mut keychain_tracker.txout_index, stop_gap);
+
+                    if keychain_tracker.txout_index.is_relevant(&tx) {
+                        update.insert_tx(tx.clone(), Some(height))?;
+                    }
+
+                    for (keychain, this_ind) in keychain_tracker.txout_index.scan(&tx) {
+                        last_active_indexes
+                            .entry(keychain)
+                            .and_modify(|ind| {
+                                if this_ind > *ind {
+                                    *ind = this_ind;
+                                }
+                            })
+                            .or_insert(this_ind);
+                    }
+                }
+
+                let wallet_scan = KeychainScan {
+                    update,
+                    last_active_indexes,
+                };
+
+                let changeset = keychain_tracker.determine_changeset(&wallet_scan)?;
+                db.append_changeset(&changeset)?;
+                keychain_tracker.apply_changeset(changeset);
```

Comment:
> I see you've avoided the need to "trim" the spks derived by only storing the last active indexes in the database. Can we add that anyway? i.e. imagine that you need to keep using the keychain_tracker after this i.e. get new address. We want it to return the one right after the last active index.

This is now done here:

https://github.com/LLFourn/bdk_core_staging/blob/80e116a27afc1653c5a920778754a98d2bbff866/bdk_keychain/src/keychain_txout_index.rs#L305-L320

> A good motivating feature would be to create a REPL mode for the cli where it does a scan and then puts you into a REPL where you can call all the cli stuff but it will also poll for new blocks/mempool txs in the background.

This will be great. Let's do that in a separate PR? #93